src/App.tsx

```diff
@@ -1,60 +1,126 @@
+import { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('0');
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+  const [allTodos, setAllTodos] = useState(todos);
+  const [addButtonClicked, setAddButtonClicked] = useState(false);
```

Comment:
I think this state [addButtonClicked, setAddButtonClicked] is unnecessary, you don't have to check if the button was clicked, because this function handleSubmit will run only after clicking on a button.

Reply:
totally agree with Klaudia 😉